src/core/tUnification.ml

```diff
@@ -630,6 +632,8 @@ let rec unify (uctx : unification_context) a b =
 	| TAbstract ({ a_path = ["haxe"],"NotVoid" },[]), _
 	| _, TAbstract ({ a_path = ["haxe"],"NotVoid" },[]) ->
 		()
+	| TAbstract _, TAbstract ({ a_path = ["haxe"],("FlatEnum" | "Function" | "Constructible") },_) ->
```

Comment:
Why did this become necessary?

Reply:
If you have an abstract over a type that passes `Constructible` constraint, this abstract itself should not unify with `Constructible`. Same for `FlatEnum` or `Function`.

Before when two abstracts were compared they did not allow any transitive casts (unless one of them is core), that's why `A -> B -> C` in original issue is not working.

Now because transitive casts are checked and as those constraint abstracts do not use usual abstract unification rules (they have their custom ones), this rule is needed.